x-pack/plugins/maps/public/actions/map_actions.ts

```diff
@@ -108,6 +115,51 @@ export function updateMapSetting(
   };
 }
 
+export function updateCustomIcons(customIcons: CustomIcon[]) {
+  return {
+    type: UPDATE_MAP_SETTING,
+    settingKey: 'customIcons',
+    settingValue: customIcons.map((icon) => {
+      return { ...icon, svg: Buffer.from(icon.svg).toString('base64') };
+    }),
+  };
+}
+
+export function deleteCustomIcon(value: string) {
+  return async (
+    dispatch: ThunkDispatch<MapStoreState, void, AnyAction>,
+    getState: () => MapStoreState
+  ) => {
+    const layersContainingCustomIcon = getLayerList(getState()).filter((layer) => {
+      const style = layer.getCurrentStyle();
+      if (!style || style.getType() !== LAYER_STYLE_TYPE.VECTOR) {
+        return;
+      }
+      return (style as IVectorStyle).getCustomIconIdsInUse().includes(value);
+    });
+    const layerList = await asyncMap(layersContainingCustomIcon, async (layer) => {
```

Comment:
Can you move gathering of layer names into block with warning since they are only needed when toast is displayed.